runtime/mgmt/mgmtinit.c

```diff
@@ -31,3 +33,48 @@ JNI_OnLoad(JavaVM *vm, void *reserved)
 {
 	return JNI_VERSION_1_8;
 }
+
+/**
+ * Determines whether file path has user permissions only.
+ *
+ * @param path file path
+ * @return Returns false on Windows and true if the file has only user permissions,
+ * false otherwise on other operation systems.
+ *
+ * @throws IOException in case of an error
+ */
+JNIEXPORT jboolean JNICALL 
+Java_sun_management_FileSystemImpl_isAccessUserOnly0(JNIEnv *env, jclass c, jstring path)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	const char* pathUTF;
+	struct J9FileStat buf;
+	I_32 status;
+	jboolean result = JNI_FALSE;
+
+	PORT_ACCESS_FROM_VMC( ((J9VMThread *) env) );
+
+	pathUTF = (*env)->GetStringUTFChars(env, path, NULL);
```

Comment:
The String needs to be converted to platform bytes before calling j9file_stat(), otherwise this will only work on Windows and when the file name is ascii and the current platform encoding is compatible with ascii.